Assets/BossRoom/Scripts/Client/Game/State/ClientMainMenuState.cs

```diff
@@ -9,6 +19,93 @@ namespace Unity.Multiplayer.Samples.BossRoom.Client
     /// </remarks>
     public class ClientMainMenuState : GameStateBehaviour
     {
-        public override GameState ActiveState { get { return GameState.MainMenu; } }
+        public override GameState ActiveState { get { return GameState.MainMenu;  } }
+
+        private GameNetPortal m_GameNetPortal;
+
+        private ClientGameNetPortal m_ClientNetPortal;
+
+
+        [SerializeField] private GameObject[] _autoInjected;
+        private DIScope m_Scope;
+
+        [SerializeField] private NameGenerationData m_NameGenerationData;
+        [SerializeField] private LobbyUIMediator m_lobbyUIMediator;
+
+        [SerializeField] private CanvasGroup m_mainMenuButtons;
+        [SerializeField] private GameObject m_signInSpinner;
+
+        private void Awake()
+        {
+            m_mainMenuButtons.interactable = false;
+            m_lobbyUIMediator.Hide();
+            DIScope.RootScope.InjectIn(this);
+        }
+
+        [Inject]
+        private void InjectDependenciesAndInitialize(AuthenticationAPIInterface authAPIInterface)
+        {
+            m_Scope = new DIScope(DIScope.RootScope);
+
+            m_Scope.BindInstanceAsSingle(m_NameGenerationData);
+            m_Scope.BindInstanceAsSingle(m_lobbyUIMediator);
+
+            var unityAuthenticationInitOptions = new InitializationOptions();
+
+#if UNITY_EDITOR
+            if (ClonesManager.IsClone())
```

Comment:
Might be worth it to add a line or two in comments as to why we're doing this block of code (I know cause we discussed it, but our readers would not)

Reply:
Done!